app/code/core/Mage/Catalog/Model/Resource/Category/Flat.php

```diff
@@ -1320,6 +1320,8 @@ public function getChildren($category, $recursive = true, $isActive = true)
         if ($isActive) {
             $select->where('is_active = ?', '1');
         }
+        $maintable = $this->getMainStoreTable($category->getStoreId());
```

Comment:
Can you move this further up and use the variable in both locations?

Reply:
Thanks for your feedback. I have refactored it.